### ISSUE TITLE
localize and internationalize

### ISSUE THREAD

**Issue by mikelmaron** [osmfoundation/welcome-mat#8]

**Comment by althio** [comment]
What is the exact plan, the roadmap here?

1. initial content in English
2. opening to OSMF members for feedback
3. new content in English
4. localize and internationalize (how?)
5. opening to OSM community for translations
6. on-hold for translations (how long?)
7. official/public announcement

> Starting point would be to create a directory with the two letter code for the language, copy all the content files and directory structure from the main page of the site there, and then translate.

You mean translators would contribute by git/Github?

**Comment by woodpeck** [comment]
In my opinion, localization and internationalization should be an optional extra and not hold back the main, English-language site. 

Don't forget, the "welcome mat" is not a general presentation of OSM to the public, it is aimed at organisations approaching OSM from various angles. Every single case in the past where I thought "I wish we already had the welcome mat" was a case where the organisation in question was using English to communicate. The larger and more influential the organisation, the more important it is they know how OSM works - but the more likely it is that they can digest English documents, too.

So, I'd say the roadmap is publish the English document after incorporating feedback, have it mature for a while, and then encourage community members to contribute translations, and make a translation public once the relevant community is happy with it.

Let's also be mindful of the fact that local chapters might have similar documents of their own; if, say, a good OSM primer for organisations already existed on openstreetmap.fr, there would be not reason for them to fall in line behind the Welcome Mat, they could just continue using what they have.

I fact, I would very much like to limit the centrally-published Welcome Mat to a small number of languages widely used in OSM, and discourage translations that we cannot realistically keep up to date.

**Comment by mikelmaron** [comment]
Would love to have translations asap, but will move forward with what we have now too.

We're planning to announce more widely on the blog this week, gather more feedback through SotM. Following that, need to put on a more permanent url (welcome.osmfoundation.org perhaps), and then from appropriate places on osm.org and osmf.org

**Comment by althio** [comment]
> @woodpeck 
In my opinion, localization and internationalization should be an optional extra and not hold back the main, English-language site.

That is the recipe for not doing any localization and internationalization. Language diversity **is** important for a worldwide project. OSMF should aim to support all people, all organisations, in all possible countries, all possible languages. In my opinion OSMF should be concerned about translations of all its contents.

> Every single case in the past where I thought "I wish we already had the welcome mat" was a case where the organisation in question was using English to communicate.

Since OSMF is only in English, people are using English. Most likely if they can't use English, they don't read you, they don't contact you. To me, that just sounds like massive selection bias.
My experience is *very* different. I have several cases where people would rather not use English.

> The larger and more influential the organisation, the more important it is they know how OSM works - but the more likely it is that they can digest English documents, too.

This is not helping the local communities and the small-to-medium companies, and even big companies in their regional markets. This is effectively a great barrier for OSM activities in areas where the predominant language is either: Arabic, Bengali, Chinese, French, Hindi, Japanese, Malay, Portuguese, Punjabi, Russian, Spanish...

> Let's also be mindful of the fact that local chapters might have similar documents of their own; if, say, a good OSM primer for organisations already existed on openstreetmap.fr, there would be not reason for them to fall in line behind the Welcome Mat, they could just continue using what they have.

They could, just give them a choice.
Anyway, openstreetmap.fr is for France, there are other countries using French but little English: maybe they would prefer the "official" OSMF Welcome Mat with community translations rather than whatever exists in France.

> I fact, I would very much like to limit the centrally-published Welcome Mat to a small number of languages widely used in OSM, and discourage translations that we cannot realistically keep up to date.

That is maybe practical from your point of view but I don't see how you select a small number of languages or why you expect translations to go out of date.

More importantly, how do you expect to _develop_ OSM if you _keep it to languages already widely used_ in OSM? To me, that is a serious case of vicious circle.

To conclude, I would just give my wishes:
- start localization as soon as possible
- start enabling community translations and make them as easy as possible
- start thinking how to translate as many languages as possible (looking for outside help, pro bono or paying)

I think this is in line with the OSMF objectives of "support, enable, help" the project and its communities "but not control".

Maybe this is not the right place to discuss. If you want me to move the discussion to osmf-talk, let me know.

**Comment by mikelmaron** [comment]
@althio you're right, this level of discussion about localization in OSMF generally is beyond scope of the ticket. @woodpeck has his opinions, but it's not the official stance of OSMF or this particular project.

What I can say -- we want to internationalize and localize the welcome mat. What can you do to help implement that right now?

**Comment by kamicut** [comment]
> We should probably wait for the content to settle a bit, before going into translations though.
@mikelmaron I'm interested in localizing for a language, do we think the content is settled?

**Comment by mikelmaron** [comment]
@kamicut yes we're stable. One thing we need to bring in is internationalization. Do you want to take a look at how it was done in https://github.com/mapbox/mapping/tree/mb-pages (which was the starting point for this site) and see about adding internationalization, and the localize?

**Comment by Andygol** [comment]
I'd like to introduce a reworked Welcome Mat featuring internationalization support and a day/night mode, all built using [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/). 

You can check out the project here: https://github.com/Andygol/welcome-mat-osmf

![image](https://github.com/osmfoundation/welcome-mat/assets/369696/1ddb140e-84d5-4643-86a4-31956e753287) | ![image](https://github.com/osmfoundation/welcome-mat/assets/369696/c8db7125-071f-4cee-9494-e8defaf8bc8f)
--- | ---

Please feel free to explore and utilize it for your needs. 

PS
Additionally, you can integrate Transifex to facilitate translations into various languages and ensure a global reach.

**Comment by angoca** [comment]
This project does seem to be very active (last commit 2 years ago), but the content is very important for welcoming non-English speakers, as well to promote community [diversity](https://osmfoundation.org/wiki/Diversity_Statement). This GitHub issue is 7 years old, and it has not too much activity.

Also, I see the proposal from @Andygol that could help this, but it seems ignored.

Let's do something, how can I collaborate?

**Comment by Andygol** [comment]
@angoca So far, you are welcome to contribute translations to any language you know to my repo. (See https://github.com/Andygol/welcome-mat-osmf#contribution)

**Comment by mikelmaron** [comment]
@Andygol how customizable is the template for Material? I have a bunch of feedback but curious about that. For example, given the primary audience of this site is not particularly technical, the GitHub UI element in the upper right would be better to include in some way in the footer.

<img width="216" alt="Image" src="https://github.com/user-attachments/assets/ceede767-1c54-457a-ad9b-c0ac74fa8eda" />

**Comment by Andygol** [comment]
@mikelmaron In case you do not want to drag much attention to the repo itself, it is possible to not show it at all (https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/)

**Comment by Andygol** [comment]
All you need to do to make the repo less noticeable is to comment/delete two lines in the settings file, and it will not be displayed.

https://github.com/Andygol/welcome-mat-osmf/blob/037a74ccfc7f94e1eb33860db3387f8be7de0866/mkdocs.yml#L7-L8

Please note that I tried to keep the original idea, and if you click on the (Ⓘ) button, you can see the relevant information and the link to the repo.

<img width="1383" alt="Image" src="https://github.com/user-attachments/assets/92a30b86-0400-4a32-b1f7-0be9cf7e9b85" />

**Comment by 1ec5** [comment]
@Andygol, @Firefishy just integrated your revamp of the site! Now the site is localized into a few languages, but let’s leave this issue open to track adopting a translation management system for the site.

**Comment by 1ec5** [comment]
@Andygol, we have a couple outstanding translation PRs. 🎉 Any tips for what to look for when reviewing them in the context of MkDocs?

**Comment by Andygol** [comment]
@1ec5 Most recommendations from “How to translate” are applicable to reviewing PRs. However, since it’s a fully manual process, one must be cautious about the `mkdocs.yml` file. Below is an example when accidentally a translated string was spoilt. 🤕 👇 

https://github.com/osmfoundation/welcome-mat/pull/136/files#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R328

It'd be great if people could avoid editing the `mkdocs.yml` file directly, and instead make changes to the relevant files in `/docs/xx/xx.yml`. Then these changes would be automatically added to the main `mkdocs.yml` file. But without using Transifex/etc., it's tricky to keep track of what people are doing in other language sections outside of their main language. If you run MkDocs locally to check your translations, you might find that the translation doesn't show up unless you enter the right translations in `mkdocs.yml`.
